imod/mf6/dis.py

```diff
@@ -234,3 +236,54 @@ def from_imod5_data(
     @classmethod
     def get_regrid_methods(cls) -> DiscretizationRegridMethod:
         return deepcopy(cls._regrid_method)
+
+    def regrid_like(
+        self,
+        target_grid: GridDataArray,
+        regrid_cache: RegridderWeightsCache,
+        regridder_types: Optional[RegridMethodType] = None,
+    ) -> Union["StructuredDiscretization", VerticesDiscretization]:
+        """
+        Regrid discretization package. Creates a StructuredDiscretization, or
+        VerticesDiscretization package, based on type of target_grid. It regrids
+        all the arrays in this package to the desired discretization, and leaves
+        the options unmodified.
+
+        The default regridding methods are specified in the ``_regrid_method``
+        attribute of the package. These defaults can be overridden using the
+        input parameters of this function.
+
+        Parameters
+        ----------
+        target_grid: xr.DataArray or xu.UgridDataArray
+            a grid defined over the same discretization as the one we want to
+            regrid the package to.
+        regrid_cache: RegridderWeightsCache, optional
```

Comment:
There is, it could use a bit of improvement though: https://deltares.github.io/imod-python/user-guide/08-regridding.html#regridding-using-non-default-methods